parcels/examples/example_globcurrent.py

```diff
@@ -183,3 +183,37 @@ class MyParticle(ptype[mode]):
     pset = ParticleSet(fieldset, pclass=MyParticle, lon=[25], lat=[-35], time=time)
 
     pset.execute(AdvectionRK4, runtime=delta(days=1), dt=dt)
+
+
+@pytest.mark.parametrize('mode', ['scipy', 'jit'])
+def test_globcurrent_particle_independence(mode, rundays=5):
+    fieldset = set_globcurrent_fieldset()
+    time0 = fieldset.U.grid.time[0]
+
+    def DeleteP0(particle, fieldset, time):
+        if particle.id == 0:
+            return ErrorCode.ErrorOutOfBounds  # we want to pass through recov loop
+
+    def DeleteParticle(particle, fieldset, time):
+        particle.delete()
+
+    pset0 = ParticleSet(fieldset, pclass=JITParticle,
+                        lon=[25, 25],
+                        lat=[-35, -35],
+                        time=time0)
+
+    pset0.execute(pset0.Kernel(DeleteP0)+AdvectionRK4,
+                  runtime=delta(days=rundays),
+                  dt=delta(minutes=5),
+                  recovery={ErrorCode.ErrorOutOfBounds: DeleteParticle})
+
+    pset1 = ParticleSet(fieldset, pclass=JITParticle,
+                        lon=[25, 25],
+                        lat=[-35, -35],
+                        time=time0)
```

Comment:
This unit test should also work without explicitly setting `time`, right? If so, would be good to remove so that that feature is also tested.

Reply:
If you don't put explicitly a time, `particle.time` is set to `fieldset.U.grid.time[0]`, but with the defer_loading, `fieldset.U.grid.time` has only 3 time steps, which are moving.

So we can either keep it as it is, or set the `particle.time` to fieldset.U.grid.time_full[0]`, which actually also exists for non defer mode (in which case it is equal to time)

The second option actually makes sense. Otherwise, it's not very clear when do we set the particle...

Reply:
Good point. I would say that it's a bug that we don't default to grid.full_time[0] for particle initialisation, as that is what users would expect (they shouldn't have to care about the three time steps in deferred_mode)

Actually, would it make sense to change the naming of `time` to e.g. `_defer_time` (signalling that it's a private attribute) and of `time_full` to `time`? A major refactoring, I know.. Perhaps better for a separate Issue?